plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -357,59 +353,71 @@ maybeAddSubst body a s n rhs = do
     hints <- view iiHints
     let hinted = shouldInline hints a n
 
-    preUnconditional <- preInlineUnconditional rhs'
-    if preUnconditional
+    if hinted -- if we've been told specifically, then do it right away
```

Comment:
Probably fine, but beware that this is not necessarily an improvement. This does mean we check `preInlineUnconditional` less, but it also means we check `hinted` more. And the inline hints can contain arbitrary logic with unknown complexity.

Reply:
> the inline hints can contain arbitrary logic with unknown complexity.

this should just be checking a bool? Can you explain a bit more?

Reply:
`shouldInline hints a n` can have arbitrary cost.

Reply:
I think it's a reasonable constraint on the caller that computing the inlining hints should not be expensive!